roles/ansible/tower/manage-credential-types/tasks/process-credential-type.yml

```diff
@@ -0,0 +1,59 @@
+---
+
+- name: "Get the credential_type id based on the name"
+  set_fact:
+    credential_type_id: "{{ item.id }}"
+  when:
+  - item.name|trim == credential_type.name|trim
+  with_items:
+  - "{{ existing_credential_types_output.rest_output }}"
+
+- name: create temporary directory to store the processed template
+  tempfile:
+    state: directory
+    suffix: cred_type
+  register: tmpdir
+
+- name: "Process Credential Type Template"
+  template:
+    src: "../templates/credential-type.j2"
+    dest: "{{ tmpdir.path }}/credential-type.json"
+  register: credential_type_json
+
+- set_fact:
+    cred_type_json_content: "{{ lookup('file','{{ credential_type_json.dest }}') }}"
+
+- name: "Load up the credential type"
+  uri:
+    url: "{{ tower_url }}/api/v2/credential_types/"
+    method: POST
+    body: "{{ cred_type_json_content }}"
+    body_format: 'json'
+    headers:
+      Content-Type: "application/json"
+      Accept: "application/json"
+    user: admin
+    password: "{{ tower_admin_password }}"
+    validate_certs: no
+    status_code: 201
+  when: credential_type_id is not defined
+
+- name: "Update existing credential type"
```

Comment:
Instead of having 2 very identical tasks, can we please just set a couple of facts up front (i.e.: `url` and `method` - and maybe `status_code`?) and hence have just one "big block" of a call to the `uri` module?

Reply:
I don't really like that idea, I think it's more simple this way and even easier to understand the code rather than adding some extra facts.